Amplify/Categories/DataStore/Model/Lazy/DefaultModelProvider.swift

```diff
@@ -24,7 +24,7 @@ public struct DefaultModelProvider<Element: Model>: ModelProvider {
     public func load() async throws -> Element? {
         switch loadedState {
         case .notLoaded:
-            throw CoreError.clientValidation("DefaultModelProvider does not provide loading capabilities", "")
+            return nil
```

Comment:
This is the main functional change. If the default model provider is called from a not loaded state, it should simply return `nil`. The caller (LazyReference) is responsible for returning `nil` on `get()` or throwing if `require()` is called.